_includes/macros/activity-snippets.html

```diff
@@ -0,0 +1,111 @@
+
+   Description:
+
+   List a bunch of activity snippets when given:
+
+   tag:            A tag to use in the query.
+
+   activity_types: An array of activity types.
+                   Possible values: posts, op-ed, press release, speech, testimony
+
+   ========================================================================== #}
+
+{% macro render(tag, activity_types=['posts', 'op-ed', 'press release', 'speech', 'testimony', 'cfpb_report']) %}
+    {% from 'macros/util/text.html' import string_length as string_length %}
+    <div class="content-l content-l__large-gutters">
+    {% for activity_type in activity_types %}
+        {% set activity = _activity_snippet(activity_type, tag, 2) %}
+        {% if string_length(activity)|int > 0 %}
+            <div class="content-l_col content-l_col-1-2">
+                {{ activity }}
+            </div>
+        {% endif %}
+    {% endfor %}
+    </div>
+{% endmacro %}
+
+
+{# ==========================================================================
+
+   _activity_snippet()
+
+   ==========================================================================
+
+   Description:
+
+   Private method to list titles of recent document type posts when given:
+
+   doctype:     The document type you wish to display.
+                Note that newsroom is not a supported document tpye in this
+                macro. Use newsroom categories instead.
+
+   querystring: The elastic search query you'd like to make.
+                Example: 'category:"Press Release" AND tags:Students'.
+                Reference: http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/query-dsl-query-string-query.html
+
+   quantity:    The number of results you want to display.
+
+   ========================================================================== #}
+
+{% macro _activity_snippet(activity_type, tags, quantity) %}
+    {% import 'macros/category-icon.html' as category_icon %}
+    {% if activity_type|lower == 'posts' %}
+        {% set doc_type = activity_type %}
+        {% set icon = category_icon.render('blog') %}
+        {% set header = 'Blog' %}
+        {% set feed = queries[doc_type].search_with_url_arguments(size=quantity,
+            filter_tags=tags) %}
+    {% elif activity_type|lower == 'op-ed' %}
+        {% set doc_type = 'newsroom' %}
+        {% set icon = category_icon.render(activity_type) %}
+        {% set header = 'Op-Ed' %}
+        {% set feed = queries[doc_type].search_with_url_arguments(size=quantity,
+            filter_category='Op-Ed', filter_tags=tags) %}
+    {% elif activity_type|lower == 'press release' %}
+        {% set doc_type = 'newsroom' %}
+        {% set icon = category_icon.render(activity_type) %}
+        {% set header = 'Press Release' %}
+        {% set feed = queries[doc_type].search_with_url_arguments(size=quantity,
+            filter_category='Press Release', filter_tags=tags) %}
+    {% elif activity_type|lower == 'speech' %}
```

Comment:
Yes! Added in https://github.com/cfpb/cfgov-refresh/commit/588f1e501da87991b158c0c4ff9732cb605372f2 but please scrutinize, I'm not 100% sure that's how it's supposed to be.